.github/workflows/smoke.yaml

```diff
@@ -0,0 +1,164 @@
+name: test
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+      - test/smoke
+  push:
+    branches:
+      - main
+      - test/smoke
+    paths-ignore:
+      - docs/**
+  workflow_dispatch:
+
+jobs:
+  check-label:
+    runs-on: ubuntu-22.04
+    outputs:
+      run_smoke_tests: ${{ steps.check.outputs.run_smoke_tests }}
+    steps:
+      - name: Check if PR author is a member of the organization or has the run-smoke label
+        id: check
+        run: |
```

Comment:
Is there a better way to gate this than running a script? Would the script cancel with an error if one of the curl requests gets something like a 500 back?

Reply:
This has been updated a bit since your initial comment. Any non `204` response code from the curl command will cause smoke tests to be skipped, but will not cause the check-labels job itself to fail. I can tinker with this some more so that the check-labels job does fail for that case if you think it's worthwhile.

Reply:
I don't think there's a need to do that. I just wanted to be as sure as possible that the tests themselves will get skipped. :+1: